modules/operations/divide.js

```diff
@@ -0,0 +1,114 @@
+// @ts-check
+import { t } from '../core/localizer';
+import { actionDivide } from '../actions/divide';
+import { behaviorOperation } from '../behavior/operation';
+import { utilGetAllNodes } from '../util';
+import { modeSelect } from '../modes/select';
+
+/** the maximum number of sections allowed, limited for performance reasons */
+const DIVIDE_LIMIT = 200;
+
+
+/** @param {string[]} selectedIDs */
+export function operationDivide(context, selectedIDs) {
+    let _extent;
+    const _action = getAction(selectedIDs[0]);
+    const _coords = utilGetAllNodes(selectedIDs, context.graph()).map(n => n.loc);
+
+    /** @param {string} entityID */
+    function getAction(entityID) {
+        const entity = context.entity(entityID);
+
+        // this operation is only show when a single way is selected
+        if (entity.type !== 'way' || selectedIDs.length !== 1) return null;
+
+        if (!_extent) {
+            _extent =  entity.extent(context.graph());
+        } else {
+            _extent = _extent.extend(entity.extent(context.graph()));
+        }
+
+        return actionDivide(entityID, context.projection);
+    }
+
+    const operation = () => {
+        if (!_action) return;
+
+        const longLength = +prompt('Long Length');
+        if (!longLength) return;
+        const shortLength = +prompt('Short Length');
```

Comment:
Unfortunately, `prompt` is not great for this, for example because language metadata cannot be supplied (see #7963). We'd need to find a better solution for this.